plugins/in_disk/in_disk.c

```diff
@@ -214,36 +214,25 @@ static int configure(struct flb_in_disk_config *disk_config,
                      struct flb_input_instance *in)
 {
     (void) *in;
-    const char *pval = NULL;
     int entry = 0;
     int i;
+    int ret;
 
-    /* interval settings */
-    pval = flb_input_get_property("interval_sec", in);
-    if (pval != NULL && atoi(pval) >= 0) {
-        disk_config->interval_sec = atoi(pval);
-    }
-    else {
-        disk_config->interval_sec = DEFAULT_INTERVAL_SEC;
-    }
-
-    pval = flb_input_get_property("interval_nsec", in);
-    if (pval != NULL && atoi(pval) >= 0) {
-        disk_config->interval_nsec = atoi(pval);
-    }
-    else {
-        disk_config->interval_nsec = DEFAULT_INTERVAL_NSEC;
+    /* Load the config map */
+    ret = flb_input_config_map_set(in, (void *)disk_config);
+    if (ret == -1) {
+        return -1;
     }
 
+    /* interval settings */
     if (disk_config->interval_sec <= 0 && disk_config->interval_nsec <= 0) {
         /* Illegal settings. Override them. */
-        disk_config->interval_sec = DEFAULT_INTERVAL_SEC;
-        disk_config->interval_nsec = DEFAULT_INTERVAL_NSEC;
+        disk_config->interval_sec = atoi(DEFAULT_INTERVAL_SEC);
+        disk_config->interval_nsec = atoi(DEFAULT_INTERVAL_NSEC);
     }
 
-    pval = flb_input_get_property("dev_name", in);
-    if (pval != NULL) {
-        disk_config->dev_name = flb_strdup(pval);
+    if (disk_config->dev_name != NULL) {
+        disk_config->dev_name = flb_strdup(disk_config->dev_name);
```

Comment:
I think this block should be removed since `disk_config->devname` will be allocated by config_map framework.
It means we don't need check `disk_config->dev_name` is NULL or not.

Reply:
`flb_free(disk_config->dev_name);` of `in_disk_exit` should be removed since it will be freed by config_map framework.